bridge-fungible-token/src/bridge_fungible_token.sw

```diff
@@ -71,31 +79,47 @@ impl MessageReceiver for Contract {
         };
 
         let res_amount = adjust_deposit_decimals(message_data.amount, DECIMALS, BRIDGED_TOKEN_DECIMALS);
+
         match res_amount {
             Result::Err(e) => {
                 // register a refund if value can't be adjusted
                 register_refund(message_data.from, message_data.token, message_data.amount);
             },
-            Result::Ok(a) => {
-                storage.tokens_minted += a;
-                mint_to_address(a, message_data.to);
+            Result::Ok(amount) => {
+                match storage.tokens_minted.try_read() {
+                    Option::Some(value) => storage.tokens_minted.write(value + amount),
+                    Option::None => storage.tokens_minted.write(amount),
+                };
+                mint_to(amount, message_data.to);
```

Comment:
If `data > 161` then we actually need to be minting to ourselves, then forwarding the coins to the contract when we call `process_message`.